checks.d/burrow.py

```diff
@@ -132,30 +136,32 @@ def _consumer_groups_offsets(self, clusters, burrow_address, extra_tags):
             consumers_path = "%s/%s/consumer" % (CLUSTER_ENDPOINT, cluster)
             consumers_list = self._rest_request_to_json(burrow_address, consumers_path).get("consumers", [])
             for consumer in consumers_list:
-                topics_path = "%s/%s/topic" % (consumers_path, consumer)
-                topics_list = self._rest_request_to_json(burrow_address, topics_path).get("topics", [])
-                for topic in topics_list:
-                    topic_path = "%s/%s" % (topics_path, topic)
-                    response = self._rest_request_to_json(burrow_address, topic_path)
-                    if not response:
-                        continue
+                topics_path = "%s/%s" % (consumers_path, consumer)
+                topics = self._rest_request_to_json(burrow_address, topics_path).get("topics", [])
+                for topic in topics:
                     tags = ["topic:%s" % topic, "cluster:%s" % cluster,
                             "consumer:%s" % consumer] + extra_tags
-                    self._submit_offsets_from_json(offsets_type="consumer", json=response, tags=tags)
+                    self._submit_offsets_from_json(offsets_type="consumer", offsets_list=topics[topic], tags=tags)
 
-    def _submit_offsets_from_json(self, offsets_type, json, tags):
+    def _submit_offsets_from_json(self, offsets_type, offsets_list, tags):
         """
         Find the offsets and push them into the metrics
         """
-        offsets = json.get("offsets")
-        if offsets:
-            # for unconsumed or empty partitions, change an offset of -1 to 0 so the
-            # sum isn't affected by the number of empty partitions.
-            offsets = [max(offset, 0) for offset in offsets]
-            self.gauge("kafka.%s.offsets.total" % offsets_type, sum(offsets), tags=tags)
-            for partition_number, offset in enumerate(offsets):
-                new_tags = tags + ["partition:%s" % partition_number]
-                self.gauge("kafka.%s.offsets" % offsets_type, offset, tags=new_tags)
+        # for unconsumed or empty partitions, change an offset of -1 to 0 so the
+        # sum isn't affected by the number of empty partitions.
+        offsets = []
+        for partition_number, offset_info in enumerate(offsets_list):
+            new_tags = tags + ["partition:%s" % partition_number]
+            latest_offset = None
+            offset_timestamp = 0
+            for offset in offset_info["offsets"]:
+                if not offset:
+                    continue
+                if offset["timestamp"] > offset_timestamp:
+                    latest_offset = offset["offset"]
```

Comment:
offsets += [latest_offset]

Reply:
hi @jrpilat , unfortunately we're still on Kafka v2. Do you mind sharing an example json body what this looks like in v3 to help us confirm if this would break v2 or not.. Thanks!

Reply:
@lenfree is this what you need?

```
curl -s http://localhost:8000/v3/kafka/default/consumer/test | jq .
```

```json
{
  "error": false,
  "message": "consumer detail returned",
  "topics": {
    "Maker": [
      {
        "offsets": [
          {
            "offset": 247136,
            "timestamp": 1571187859717,
            "lag": 432
          },
          {
            "offset": 247175,
            "timestamp": 1571187860801,
            "lag": 393
          },
          {
            "offset": 247203,
            "timestamp": 1571187861806,
            "lag": 365
          },
          {
            "offset": 247254,
            "timestamp": 1571187862816,
            "lag": 314
          },
          {
            "offset": 247285,
            "timestamp": 1571187863866,
            "lag": 283
          },
          {
            "offset": 247323,
            "timestamp": 1571187865134,
            "lag": 245
          },
          {
            "offset": 247378,
            "timestamp": 1571187866144,
            "lag": 190
          },
          {
            "offset": 247438,
            "timestamp": 1571187867176,
            "lag": 130
          },
          {
            "offset": 247462,
            "timestamp": 1571187868263,
            "lag": 106
          },
          {
            "offset": 247492,
            "timestamp": 1571187869406,
            "lag": 76
          },
          {
            "offset": 247504,
            "timestamp": 1571187871205,
            "lag": 64
          },
          {
            "offset": 247516,
            "timestamp": 1571187872322,
            "lag": 52
          },
          {
            "offset": 247556,
            "timestamp": 1571187873389,
            "lag": 12
          },
          {
            "offset": 247566,
            "timestamp": 1571187875598,
            "lag": 2
          },
          {
            "offset": 247568,
            "timestamp": 1571187978098,
            "lag": 0
          }
        ],
        "owner": "",
        "client_id": "",
        "current-lag": 0
      }
    ]
  },
  "request": {
    "url": "/v3/kafka/default/consumer/test",
    "host": "757320aea5d7"
  }
}
```